dali/test/python/reader/test_fits.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali import Pipeline
+import nvidia.dali.fn as fn
+import os
+from astropy.io import fits
+import numpy as np
+import tempfile
+import random
+from test_utils import to_array
+from numpy.testing import assert_array_equal
+
+rng = np.random.RandomState(12345)
+
+
+def create_fits_file(filename, shape, type=np.int32, compressed=False):
+    imageData = rng.randint(100, size=shape).astype(type)
+    imageHeader = fits.Header()
+    hdu = fits.ImageHDU(imageData, imageHeader)
+    if compressed:
+        hdu = fits.CompImageHDU(imageData, imageHeader)
+
+    hdu.writeto(filename)
+
+
+def delete_fits_file(filename):
+    if os.path.isfile(filename):
+        os.remove(filename)
+
+
+def FitsReaderPipeline(path, batch_size, device="cpu", file_list=None, files=None,
+                       file_filter="*.fits", num_threads=1, device_id=0,
+                       hdu_indices=None, dtype=None):
+    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=device_id)
+    data = fn.readers.fits(device=device,
+                           file_list=file_list,
+                           files=files,
+                           file_root=path,
+                           file_filter=file_filter,
+                           shard_id=0,
+                           num_shards=1)
+    pipe.set_outputs(data)
+    return pipe
+
+
+all_numpy_types = set([
+    np.bool_, np.byte, np.ubyte, np.short, np.ushort, np.intc, np.uintc, np.int_, np.uint,
+    np.longlong, np.ulonglong, np.half, np.float16, np.single, np.double, np.longdouble, np.csingle,
+    np.cdouble, np.clongdouble, np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16,
+    np.uint32, np.uint64, np.intp, np.uintp, np.float32, np.float64, np.float_, np.complex64,
+    np.complex128, np.complex_
+])
```

Comment:
Let's limit ourselves to some subset of the image types supported by FITS:
u8, i16, i32, i64, u32, u64.